drawBot/context/baseContext.py

```diff
@@ -886,6 +875,58 @@ def copy(self):
         return new
 
 
+def makeTextBoxes(attributedString, xy, align):
+    extraPadding = 20
+    x, y = xy
+    w, h = attributedString.size()
+    w += extraPadding
+
+    if align is not None:
+        attributedString = attributedString.mutableCopy()
+        # overwrite all align settings in each paragraph style
+        def block(value, rng, stop):
+            value = value.mutableCopy()
+            value.setAlignment_(FormattedString._textAlignMap[align])
+            attributedString.addAttribute_value_range_(AppKit.NSParagraphStyleAttributeName, value, rng)
+        attributedString.enumerateAttribute_inRange_options_usingBlock_(AppKit.NSParagraphStyleAttributeName, (0, len(attributedString)), 0, block)
```

Comment:
Wouldn't `attributedString.addAttribute_value_range_(AppKit.NSParagraphStyleAttributeName, value, rng)` over the whole range without enumerating be working, too?

Reply:
no, cause a NSParagraphStyle can contain different kinds of settings. It could be that  a formattedString contains different line heights... so there are multiple paragraph styles

Reply:
Ahh, duh, I missed the paragraph style object, got it now. Alignment is not a single simple attr.